### PR TITLE
UI/UX of relations

### PR DESCRIPTION
**Pull request by tomasMizera**
Enhanced UI and UX of relations widget.

Added _More_ and _Add_ icons for text mode and _Add_ for photo mode.

Initial state of text mode saying _n linked features_ was deleted. It now expands immediately. Also changed how the features look in widget.

Sorts out all points from #1530 but one about notification when feature could not be saved.

https://user-images.githubusercontent.com/22449698/124290302-8d0add00-db53-11eb-80cf-9306ce142547.mp4

Also checkbox for _Reuse last value option_ is now forbidden to show with relation.

### CODE REVIEW

File: app/qml/editor/RelationPhotoDelegate.qml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      if (image.status === Image.Error) {
+        image.imageValid = false
+        customStyle.icons.notAvailable
+      }
+      else if (absolutePath !== '' && __inputUtils.fileExists(absolutePath)) {
+        "file://" + absolutePath
+      }
+      else if (absolutePath === '' || absolutePath === undefined) {
+        image.imageValid = false
+        customStyle.icons.notAvailable
```
Comment (PeterPetrik) [inline app/qml/editor/RelationPhotoDelegate.qml:53]: would fall too else branch, no?

File: app/qml/editor/inputrelation.qml [hunk 2]
```diff
@@ -189,9 +154,45 @@ Item {
...
+          backgroundContent.border.color: customStyle.relationComponent.tagBorderColorButton
+          textContent.color: customStyle.relationComponent.tagTextColor
+
+          onClicked: textModeContainer.state = "page"
+        }
+
+        RelationTextDelegate {
+          id: addChildButton
+
+          text: qsTr( "+ Add" )
```
Comment (PeterPetrik) [inline app/qml/editor/inputrelation.qml:180]: maybe "+" shouldn't be in translate text "+ " + qsTr( "Add" )

File: app/qml/form/FeatureFormStyling.qml [hunk 3]
```diff
@@ -98,5 +97,26 @@ QtObject {
 
   property QtObject relationComponent: QtObject {
     property real textDelegateHeight: fields.height * 0.8
+    property int flowSpacing: 8 * QgsQuick.Utils.dp
+
+    // photo mode
```
Comment (PeterPetrik) [inline app/qml/form/FeatureFormStyling.qml:102]: it is also possible to add a new photo?

**Review by PeterPetrik — APPROVED** [review]
haven't tested but code looks good

File: app/qml/form/FeatureFormStyling.qml (hunk 3 above)
Comment (tomasMizera) [reply app/qml/form/FeatureFormStyling.qml:102]: Yes it is

Comment (tomasMizera) [reply app/qml/form/FeatureFormStyling.qml:102]: It is in the attached video

File: app/qml/editor/RelationPhotoDelegate.qml (hunk 1 above)
Comment (tomasMizera) [reply app/qml/editor/RelationPhotoDelegate.qml]: Else branch did the same as this one so I removed this redundant check

**Pull request merged**